### PR TITLE
Semaphore 2.0

### PR DESCRIPTION
**Pull request by LionelLindt**

### CODE REVIEW

File: templates/.semaphore/develop-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version: v1.0
+name: Heroku deployment
```
Comment (coorasse) [inline templates/.semaphore/develop-deploy.yml]: what about `develop-deploy`?

File: templates/.semaphore/develop-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: Deploy
```
Comment (coorasse) [inline templates/.semaphore/develop-deploy.yml]: what about `develop-deploy`?

File: templates/.semaphore/develop-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    task:
+      secrets:
+        - name: [project-name]
+      env_vars:
+        - name: HEROKU_REMOTE
+          value: https://git.heroku.com/[project-name]-develop.git
+      jobs:
+        - name: Push code
```
Comment (coorasse) [inline templates/.semaphore/develop-deploy.yml]: what about `develop-deploy`?

File: templates/.semaphore/semaphore.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+version: "v1.0"
+name: Pipeline with parallel blocks and dependencies
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]:
```suggestion
name: [project-name]
```

File: templates/.semaphore/semaphore.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+auto_cancel:
+  running:
+    when: "true"
+blocks:
+  - name: Install dependencies
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]:
```suggestion
  - name: cache
```

File: templates/.semaphore/semaphore.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    dependencies: []
+    task:
+      secrets:
+        - name: [project-name]
+      jobs:
+        - name: cache
+          commands:
+            - checkout
+            - cache restore bundle-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock),bundle-$SEMAPHORE_GIT_BRANCH,bundle-master
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]: this is done by simply writing `cache restore`

File: templates/.semaphore/semaphore.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            - bundle install --deployment
+            - cache store bundle-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock) vendor/bundle
+            - cache store bundleconfig-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock) .bundle
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]: these two are done with simply `cache store`

File: templates/.semaphore/semaphore.yml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            - cache restore yarn-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock),yarn-$SEMAPHORE_GIT_BRANCH,yarn-master
+            - cache restore nodemodules-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock),nodemodules-$SEMAPHORE_GIT_BRANCH,nodemodules-master
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]: these are also managed by the previous `cache restore` command

File: templates/.semaphore/semaphore.yml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+            - nvm install
+            - bin/yarn install --cache-folder ~/.cache/yarn
+            - cache store yarn-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock) ~/.cache/yarn
+            - cache store nodemodules-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock) node_modules
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]: and these two by the previous `cache store`, which should be moved here.

File: templates/.semaphore/semaphore.yml [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  - name: "Tests"
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]:
```suggestion
  - name: tests
```

File: templates/.semaphore/semaphore.yml [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    dependencies: ['Install dependencies']
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]:
```suggestion
    dependencies: ['cache']
```

File: templates/.semaphore/semaphore.yml [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+          value: postgresql://postgres@localhost/test?encoding=utf8
+        - name: RAILS_ENV
+          value: test
+      prologue:
+        commands:
+          - checkout
+          - cache restore bundle-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock),bundle-$SEMAPHORE_GIT_BRANCH,bundle-master
+          - cache restore bundleconfig-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock)
+          - cache restore yarn-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock),yarn-$SEMAPHORE_GIT_BRANCH,yarn-master
+          - cache restore nodemodules-$SEMAPHORE_GIT_BRANCH-$(checksum package-lock.json),node-modules-$SEMAPHORE_GIT_BRANCH,node-modules-master
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]: `cache restore`

File: templates/.semaphore/semaphore.yml [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+          - nvm use
+          - sem-service start postgres
+          - cp config/application.example.yml config/application.yml
+          - bin/rails db:create db:schema:load
+          - bin/rails assets:precompile
+      jobs:
+        - name: "Tests"
```
Comment (coorasse) [inline templates/.semaphore/semaphore.yml]:
```suggestion
        - name: tests
```

File: ruby_on_rails/configure_cd.md [hunk 14]
```diff
@@ -4,26 +4,36 @@ We use Semaphore to deploy most of our apps. Some projects use CircleCI and
 mobile apps use Fastlane. This article is about how to deploy to Heroku with
 Semaphore.
 
-## Prerequisites
+## Setup `master`, `develop` and `testing`
 
-* Heroku API key of `admin@renuo.ch` (*wg-operations* can help with that)
+* Add the following to your `semaphore.yml`:
```
Comment (schmijos) [inline ruby_on_rails/configure_cd.md]: Where is this file located?

File: templates/.semaphore/develop-deploy.yml [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: Deploy
+    task:
+      secrets:
+        - name: [project-name]
```
Comment (schmijos) [inline templates/.semaphore/develop-deploy.yml]: Didn't we say that we have one Renuo Heroku secret for all apps?

File: templates/.semaphore/semaphore.yml (hunk 6 above)
Comment (schmijos) [reply templates/.semaphore/semaphore.yml]: But only if the gems are installed to inside the working directory. This has to be done like this on line 21:

    bundle install --deployment -j 4 --path vendor/bundle

File: ruby_on_rails/configure_cd.md (hunk 14 above)
Comment (coorasse) [inline ruby_on_rails/configure_cd.md]:
```suggestion
* Add the following file: `.semaphore/semaphore.yml`:
```

Comment (LionelLindt) [reply ruby_on_rails/configure_cd.md]: It is located under the `.semaphore` directory in the project root, as [described here](https://github.com/renuo/applications-setup-guide/blob/semaphore-2.0/ruby_on_rails/configure_ci.md#L16)

File: templates/.semaphore/semaphore.yml [hunk 16]
```diff
@@ -40,16 +35,14 @@ blocks:
       prologue:
         commands:
           - checkout
-          - cache restore bundle-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock),bundle-$SEMAPHORE_GIT_BRANCH,bundle-master
-          - cache restore bundleconfig-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock)
-          - cache restore yarn-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock),yarn-$SEMAPHORE_GIT_BRANCH,yarn-master
-          - cache restore nodemodules-$SEMAPHORE_GIT_BRANCH-$(checksum package-lock.json),node-modules-$SEMAPHORE_GIT_BRANCH,node-modules-master
+          - cache restore
+          - bundle install --deployment --path vendor/bundle
```
Comment (LionelLindt) [inline templates/.semaphore/semaphore.yml]: [This seems to be necessary:](https://docs.semaphoreci.com/article/99-rails-continuous-integration)
image

**Pull request merged**